src/components/LastVisited/withLastVisited.js

```diff
@@ -29,12 +29,31 @@ function withLastVisited(WrappedComponent) {
       });
 
       this.cachePreviousUrl = this.cachePreviousUrl.bind(this);
+      this.previousModules = [];
       this.lastVisited = {};
       this.previous = {};
 
       history.listen((location) => {
         const module = this.getCurrentModule(location);
         if (!module) return;
+
+        // When switching modules, clear the record cache older than previous module to keep memory from growing unbounded. Refs UIIN-687.
+        if (this.previousModules.length > 1) {
+          if (this.previousModules.indexOf(module.module) < 0) {
+            this.props.stripes.store.dispatch({
+              type: '@@stripes-connect/PRUNE',
+              meta: {
+                resource: 'records',
```

Comment:
I would add that `ui-data-import` in settings for file extensions, job profiles, mapping profiles uses other record keys. It is done to maintain more than one resource and thus S&S functionality per module. So, in this case, their data won't be pruned.

https://github.com/folio-org/ui-data-import/blob/master/src/settings/FileExtensions/FileExtensions.js#L45

Reply:
I don't actually believe the `meta` is doing anything, but fwiw, the `records` in this case are the [`props.resources.foo.records`](https://github.com/folio-org/stripes-connect/pull/102/files#diff-0750d38c98cb1579ed9274950cd696edR51). So I believe that anything that uses `RESTResource` will be pruned correctly, not just those which are explicitly keyed in the manifest as `records`.